### PR TITLE
support bit type

### PR DESCRIPTION
**Pull request by siddontang**
support bit type like `select 0b01, b'01'`.

see https://dev.mysql.com/doc/refman/5.7/en/bit-type.html

### CODE REVIEW

File: mysqldef/bit_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	}{
+		{"0b01", 8, 1, "0b00000001", "\x01"},
+		{"0b111111111", 16, 511, "0b0000000111111111", "\x01\xff"},
+		{"0b01", -1, 1, "0b00000001", "\x01"},
+	}
+
+	for _, t := range tbl {
+		b, err := ParseBit(t.Input, t.Width)
+		c.Assert(err, IsNil)
+		c.Assert(err, IsNil)
```
Comment (qiuyesuifeng) [inline mysqldef/bit_test.go]: Remove this line.

File: mysqldef/bit.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+func (b Bit) ToNumber() float64 {
+	return float64(b.Value)
+}
+
+// ToString returns the binary string for bit type.
+func (b Bit) ToString() string {
+	byteSize := (b.Width + 7) / 8
+	buf := make([]byte, byteSize)
+	for i := byteSize - 1; i >= 0; i-- {
+		buf[len(buf)-i-1] = byte(b.Value >> uint(i*8))
```
Comment (qiuyesuifeng) [inline mysqldef/bit.go]: Here len(buf) is equal to byteSize?

Comment (siddontang) [inline mysqldef/bit.go]: yes

File: mysqldef/bit.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+// e.g, if bit string is 0b01, the above will return 8 for its bit width.
+func ParseBit(s string, width int) (Bit, error) {
+	if len(s) == 0 {
+		return Bit{}, errors.Errorf("invalid empty string for parsing bit type")
+	}
+
+	if s[0] == 'b' || s[0] == 'B' {
+		// format is b'val' or B'val'
+		s = strings.Trim(s[1:], "'")
+	} else if strings.HasPrefix(s, "0b") {
```
Comment (qiuyesuifeng) [inline mysqldef/bit.go]: Need process "0B"?

Comment (siddontang) [inline mysqldef/bit.go]: 0B is invalid.

File: mysqldef/bit.go (hunk 2 above)
Comment (qiuyesuifeng) [inline mysqldef/bit.go]: So it is better to use `buf[byteSize-i-1] `.

**Comment by siddontang** [comment]
PTAL @shenli

**Comment by shenli** [comment]
AFAIK we should handle all types in flatten()/unflatten() in tables.go. Should we handle type hex/bit there?

**Comment by siddontang** [comment]
no, don't need. 

but I will handle hex/bit in codec later.

**Comment by siddontang** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**